pyimager/pyimager.py

```diff
@@ -5,53 +5,59 @@
 from scipy.ndimage.filters import convolve
 
 
-def circropper(input_path, output_path, margin):
+def circropper(input_path, margin):
     """
     Crops an image into a circle and leave some margin as you defined 
 
     Parameters:
     -----------
     input_path: string 
         The file path of the image to be cropped
-    output_path: string
-        The file path for cropped image 
-    margin: float
+    margin: float or int 
         The distance between circle boundary and the original image boundary
     
     Returns:
     --------
-        A numpy array of cropped image
+        A new cropped Image object 
         
     Examples:
     ---------
     >>> from pyimager import circropper
-    >>> circropper("bear.jpg", "result.png", 0)
-    A file named "result.png" will be generated in the 
-    designated folder.
+    >>> circropper("bear.jpg", 0)
+    A Image is cropped to a circle with no margin 
     """
+    # Test argument  
+    if type(input_path) != str and type(margin) != float and type(margin) != int:
+        raise TypeError("The 'input_path' argument must be a string and the margin argument must be a number")
+    if type(input_path) != str:
+        raise TypeError("The 'input_path' argument must be a string")
+    if type(margin) != float and type(margin) != int:
+        raise TypeError("The 'margin' argument must be a float")
+    
```

Comment:
Remove redundant test cases
```suggestion
    if type(input_path) != str:
        raise TypeError("The 'input_path' argument must be a string")
    if type(margin) != float and type(margin) != int:
        raise TypeError("The 'margin' argument must be of type 'float' or 'integer'")
    
```

Reply:
I think they are not redundant tests. What I want to do is when I have both arguments wrong, the system shows both "input_path" and "margin" are wrong; and when only one argument is wrong, the system shows one argument is wrong. I have tested this three conditions last night so I think these tests are fine.

Reply:
for `if type(margin) != float and type(margin) != int` though, you may want `if type(margin) != float or type(margin) != int`. For example if margin is float, statement `and` would raise error which should not.

Reply:
I don't think so because of `not` on top of each statement. Only when two conditions are `False`, both `not False` are Ture, then the error shows.

Reply:
when two conditions are false means type(margin)=float and type(margin)=int. Coding might coerce the type? I'm not sure, but logically it's impossible

Reply:
might something wrong in my previous description. But `if type(margin) != float and type(margin) != int` returns `True` only when I input a string or other type that are considered as wrong types, then the error raise. I have tried yesterday, there should be nothing wrong. You can try it then.

Reply:
these tests can pass `pytest`, no I think nothing to worry abouts